### PR TITLE
Support webpack 2

### LINKED ISSUE #146: Webpack 2 support?
Could you add support for Webpack 2?

### PR DESCRIPTION
**Pull request by aaronjensen**
This should support webpack 2 out of the box.

Fixes #146

### CODE REVIEW

**Comment by aaronjensen** [comment]
This doesn't quite work yet, there seems to be a problem w/ the loader. Looking into it.

**Comment by aaronjensen** [comment]
Ok, this should be good to go.

**Comment by sapegin** [comment]
Thanks! Does it still work in Webpack 1?

**Comment by aaronjensen** [comment]
Yep as far as I can tell. It'd be good if you gave it a test as well.

File: src/make-webpack-config.js [hunk 1]
```diff
@@ -38,22 +38,29 @@ module.exports = function(env) {
...
-			modulesDirectories: [
+			modules: [
+				__dirname,
 				path.resolve(__dirname, '../node_modules'),
 				'node_modules',
 			],
 			alias: {
-				codemirror: codeMirrorPath,
+				'codemirror': codeMirrorPath,
+				'rsg-components': path.resolve(__dirname, 'rsg-components'),
```
Comment (sapegin) [inline src/make-webpack-config.js]: This line breaks component replacement with aliases like:

```js
webpackConfig.resolve.alias['rsg-components/Wrapper'] = path.resolve(__dirname, 'lib/styleguide/Wrapper');
```

File: src/make-webpack-config.js [hunk 2]
```diff
@@ -38,22 +38,29 @@ module.exports = function(env) {
 			filename: 'build/bundle.js',
 		},
 		resolve: {
-			root: __dirname,
```
Comment (sapegin) [inline src/make-webpack-config.js]: I had to revert this line to make my style guide work.

File: src/make-webpack-config.js (hunk 1 above)
Comment (aaronjensen) [inline src/make-webpack-config.js]: Can you use `NormalModuleReplacementPlugin` for this instead?

Is this why you're not resolving `rsg-components` relatively? That was the biggest problem I had upgrading to webpack2, I'm not sure how to do that without using this alias, the new `modules` doesn't seem to actually work the same as `root` did.

**Comment by aaronjensen** [comment]
@sapegin thank you for testing. I believe I've resolved both of the issues you pointed out, could you try again please?

File: src/make-webpack-config.js [hunk 3]
```diff
@@ -38,9 +38,16 @@ module.exports = function(env) {
...
 			extensions: ['', '.js', '.jsx'],
-			modulesDirectories: [
+			// Webpack 1
+			root: __dirname,
+			moduleDirectories: [
+				path.resolve(__dirname, '../node_modules'),
```
Comment (aaronjensen) [inline src/make-webpack-config.js]: For what it's worth, I think that this actually belongs in `root` on webpack1. See: https://github.com/webpack/webpack/issues/472#issuecomment-55706013

**Comment by sapegin** [comment]
Tested again and now it just works with Webpack 1. Merging and thanks again!

**Comment by sapegin** [comment]
And this new squash and merge feature is very nice: I don’t have to ask contributors to do it manually.

**Pull request merged**